### DISCUSSION TITLE
Welcome to chess-tuning-tools Discussions!

### DISCUSSION THREAD

**Discussion by kiudee** [kiudee/chess-tuning-tools#126] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by musketeerchess** [comment]
Hi
I’m Zied, a chess fun and chess variants inventor.

The game i’m most proud of is named Musketeer Chess. It’s a Chess Game
where 2 additional pieces are added to the classic chess set.
The game is played on an 8x8 board. The additional pieces are introduced by
a drop mechanism.
When one of the pieces on 1st or 8th row move for the first time, the
additional pieces can be dropped on the square freed by the piece that
moved. The drop is not random and is predefined by both players before the
game begins.

My main goal is to deliver for the chess variants funs a
Clear methodology helping them assess the relative value of these chess
pieces.

**Reply by kiudee** [reply]
That indeed sounds like a fun variant to play.

One way I can think of to get the relative value of the chess pieces using chess-tuning-tools would be the following.
If you have an engine playing your variant, you could let it play against itself and tune the piece values like so:
```json
	"parameter_ranges": {
		"PieceValue1": "Real(0.1, 10.0)",
		"PieceValue2": "Real(0.1, 10.0)",
		"PieceValue3": "Real(0.1, 10.0)",
                ...
	},
```
The values it converges to are those maximizing the playing strength against itself.

Since chess-tuning-tools uses `cutechess-cli` to run the matches, you would need to check if your engine works with that (and if the variant causes any problems). It may be necessary to create a small fork of cutechess-cli for your variant, if not.
Other than that it suffices that your engine supports the UCI protocol and the tuner can set the parameters using `setoption`.

Another consideration is openings. It is best to use a wide opening book for tuning, otherwise the tuning process runs the risk of converging to a spurious optimum, which is overfitting the particular opening position. An alternative is to give it an empty opening book and simply apply randomization for the first n moves inside the engine.
